apps/system/lockscreen/js/lockscreen_inputpad.js

```diff
@@ -163,7 +167,19 @@
       var span = this.passcodeCode.childNodes[i];
       if (span) {
         if (this.states.passCodeEntered.length > i) {
+        var value = this.states.passCodeEntered.substr(i, this.states.passCodeEntered.length);
+        if ((this.states.passCodeEntered.length - 1 === i) && (keyCode && (keyCode !== 'b'))) {
```

Comment:
It looks like you're passing the keyCode just as some kind of flag for backspace maybe?  I'm not convinced you need this, but if you do it would be clearer to pass a boolean backspace argument.  Or just handle backspacing somewhere else.